### ISSUE TITLE
"Error initializing security header generator" on Kiwi browser (android)

### ISSUE THREAD

**Issue by DannyKpo** [dimdenGD/OldTwitter#843]
**Describe the bug**
So i'm getting that error but I don't know what setting should I change on kiwi browser, I have already disabled ublock origin as well as request "Do not track". 

**Browser**
Kiwi Browser 124.0.6327.2

**Links**
any twitter.com/ x.com link

**OldTwitter version**
1.8.5
![Screenshot_20240614-005852_Kiwi Browser](https://github.com/dimdenGD/OldTwitter/assets/70122703/95be3a72-d65d-46aa-84d5-6ee80082cc02)

**Comment by Volodesi** [comment]
Getting this as well on Thorium browser for Desktop.

![image](https://github.com/dimdenGD/OldTwitter/assets/36998033/46981e32-a134-424f-9515-4c0d9d54ff5a)

**Comment by lololq0** [comment]
just disable

> Getting this as well on Thorium browser for Desktop.
> 
> ![image](https://private-user-images.githubusercontent.com/36998033/339681712-46981e32-a134-424f-9515-4c0d9d54ff5a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgzNTE0NTIsIm5iZiI6MTcxODM1MTE1MiwicGF0aCI6Ii8zNjk5ODAzMy8zMzk2ODE3MTItNDY5ODFlMzItYTEzNC00MjRmLTk1MTUtNGMwZDlkNTRmZjVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE0VDA3NDU1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI3YzRhYWY5MjQ1NDc1ZTc0ZDMxNDYyMzIyNTc3NjljMjMxMTczNDRjZmViNzBiNGZiYTQ5MGQ5MTEzMjU1NjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Nh2t6F7nMsASCk8zRRQgyXdHFY0HJdcBlRi8Ezc3omY)


Just diablethe extension first then login.

**Comment by jdanny2002** [comment]
> > Getting this as well on Thorium browser for Desktop.
> > ![image](https://private-user-images.githubusercontent.com/36998033/339681712-46981e32-a134-424f-9515-4c0d9d54ff5a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgzNTE0NTIsIm5iZiI6MTcxODM1MTE1MiwicGF0aCI6Ii8zNjk5ODAzMy8zMzk2ODE3MTItNDY5ODFlMzItYTEzNC00MjRmLTk1MTUtNGMwZDlkNTRmZjVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE0VDA3NDU1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI3YzRhYWY5MjQ1NDc1ZTc0ZDMxNDYyMzIyNTc3NjljMjMxMTczNDRjZmViNzBiNGZiYTQ5MGQ5MTEzMjU1NjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Nh2t6F7nMsASCk8zRRQgyXdHFY0HJdcBlRi8Ezc3omY)
> 
> Just disable the extension first then login then open the extension again

already tried, as well as deleting cookies. that doesn't work. it keeps saying the same error

**Comment by lololq0** [comment]
Weird. That's how I resolved mine.

**Comment by Volodesi** [comment]
> > Getting this as well on Thorium browser for Desktop.
> > ![image](https://private-user-images.githubusercontent.com/36998033/339681712-46981e32-a134-424f-9515-4c0d9d54ff5a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgzNTE0NTIsIm5iZiI6MTcxODM1MTE1MiwicGF0aCI6Ii8zNjk5ODAzMy8zMzk2ODE3MTItNDY5ODFlMzItYTEzNC00MjRmLTk1MTUtNGMwZDlkNTRmZjVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE0VDA3NDU1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI3YzRhYWY5MjQ1NDc1ZTc0ZDMxNDYyMzIyNTc3NjljMjMxMTczNDRjZmViNzBiNGZiYTQ5MGQ5MTEzMjU1NjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Nh2t6F7nMsASCk8zRRQgyXdHFY0HJdcBlRi8Ezc3omY)
> 
> Just disable the extension first then login then open the extension again


Clicking Ok just loads Twitter like normal. So why would I disable it? It just adds an extra step when I want to visit the site.

**Comment by lololq0** [comment]
I mean for me it logs out after a long time and that error pops up. The rn I really don't have to do it anymore. I only did it once or twice then yeah.

**Comment by J6833** [comment]
I'm getting the same error.

Version 1.8.5

Opera GX
LVL5 (core: 109.0.5097.130)
Update stream:Early Access
System:Windows 11 64-bit
Chromium version:123.0.6312.124

**Comment by J6833** [comment]
This seems to be an issue with Twitter too? Saw this post from a staff member saying that they're looking into it.

Post: https://devcommunity.x.com/t/there-is-no-request-token-for-this-page-this-is-a-special-key-required-to-verify-that-the-application-uses-your-twitter-account/220914/5

**Comment by dimdenGD** [comment]
This is unrelated to Twitter. Also you can ignore the message on Kiwi Browser if page loads fine afterwards, not sure why it appears in such cases since it's only meant to appear when page is stuck loading

**Comment by 70624290** [comment]
### tl; dr: firefox and other browsers block the necessary domains for old twitter to work, add `abs.twimg.com` to the `urlclassifier.features.socialtracking.skipURLs` key in about:config as a workaround

i think i finally figured out the answer. firefox, along with other privacy-oriented browsers, all seem to use a blocklist which includes `twimg.com` and every one of its subdomains.
for some reason, in firefox it seems like requests made in a non-authoritative context, such as requests made in extensions or webpages, get filtered through a blocklist regardless of the user's preference. at least in firefox, this filtering happens at a level low enough where the request is not even logged in the network tab of the developer tools. this made debugging the error almost impossible since fetch error messages are deliberately made as vague as possible in order to minimize attack vectors. 

the only way i managed to get more details was adding a `webRequest.onErrorOccurred` listener to every request. this way, i managed to figure out that firefox was blocking requests to `twimg.com` because it was included in a list of social media trackers.

to fix this, i simply `abs.twimg.com` to the whitelist in the `urlclassifier.features.socialtracking.skipURLs` key in about:config.

currently not sure how to fix this issue in other browser, but it is almost certainly related to tracking protection features so maybe try disabling them.

**Comment by scarlion1** [comment]
Good find @70624290!  Can confirm that works.  I didn't have a `urlclassifier.features.socialtracking.skipURLs` key so created a new one (String) and added `abs.twimg.com`, refreshed Twitter and voila! it loaded...

but (to everyone) what the heck is OldTwitter doing now?  I let it load the home page and let it sit for several minutes, it ate 25% of my cpu's (1 of 4 threads) the whole time.  This is the sort of usage I see when there are animations on the screen (since it's an ancient laptop with integrated video), but nothing was animated.  As soon as I switched to another tab, such as this one on GitHub, thus taking away anything on the screen from OldTwitter, cpu dropped to near nothing again.  Then I tried forcing new twitter to load with https://x.com/home?newtwitter=true and after several seconds it had finished and the cpu usage dropped back to nothing again.  Fortunately, I hardly use Twitter, but this inefficient resource usage is enough for me to shelf OldTwitter in the mean time.  Any ideas wat causes this high usage?
